File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int n = cardPoints.size();
        vector<int>left(k+1,0),right(k+1,0); 
        for(int i=1;i<=k;i++){
            left[i] = left[i-1] + cardPoints[i-1];
            right[k-i] = right[k+1-i] + cardPoints[n-i];
        }
        int ans = 0;
        for(int i=0;i<=k;i++){
            ans = max(ans,left[i]+right[i]);
        }
        return ans;
    }
};

File: 1423-maximum-points-you-can-obtain-from-cards/1423-maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        n = len(cardPoints)
        left = [0] *(k+1)
        right = [0] *(k+1)
        for i in range(1,k+1):
            left[i] = left[i-1] + cardPoints[i-1]
            right[k-i] = right[k+1-i] + cardPoints[n-i]
        ans = 0
        for i in range(k+1):
            ans = max(ans,left[i]+right[i])
        return ans